### PR TITLE
DCOS-15173: Fix incorrect running task count

### PR DESCRIPTION
**Pull request by jfurrow**
This PR...
* introduces a `getTaskCount` method on the `Service` and `ServiceTree` structs which tallies the number of tasks as reported by Marathon
* implements the `getTaskCount` method for the `ServiceTable` and `ServiceBreadcrumbs` components

Before:

After:

**Checklist**
- [x] Did you add a JIRA issue in a commit message or as part of the branch name?
- [x] Did you add new unit tests?
- [ ] Did you add new integration tests?
- [ ] If this is a regression, did you write a test to catch this in the future?

### CODE REVIEW

**Comment by jfurrow** [comment]
Sorry @weblancaster I pushed one more commit based on feedback from @leemunroe.

File: plugins/services/src/js/components/ServiceBreadcrumbs.js [hunk 1]
```diff
@@ -121,21 +121,27 @@ class ServiceBreadcrumbs extends React.Component {
...
     const instancesCount = service.getInstancesCount();
     const isDeploying = serviceStatus === "Deploying";
 
     if (instancesCount === 0) {
       return null;
     }
 
+    let taskCountDetails = `(${runningTasksCount})`;
+
+    if (runningTasksCount !== instancesCount) {
```
Comment (weblancaster) [inline plugins/services/src/js/components/ServiceBreadcrumbs.js]: Works great @jfurrow !

What do you think about using ternary? I believe less explicit conditions keep the code cleaner. not a big deal though.
```
const taskCountDetails = runningTasksCount !== instancesCount
      ? `(${runningTasksCount} of ${instancesCount})`
      : `(${runningTasksCount})`;
```

File: plugins/services/src/js/containers/services/ServicesTable.js [hunk 2]
```diff
@@ -266,13 +266,14 @@ class ServicesTable extends React.Component {
...
-
+    const tasksRunning = service.getTaskCount();
     const isDeploying = serviceStatus === "Deploying";
 
-    const conciseOverview = ` (${tasksRunning}/${instancesCount})`;
+    let conciseOverview = ` (${tasksRunning})`;
     let verboseOverview = ` (${tasksRunning} ${StringUtil.pluralize("Instance", tasksRunning)})`;
+
     if (tasksRunning !== instancesCount) {
+      conciseOverview = ` (${tasksRunning}/${instancesCount})`;
```
Comment (weblancaster) [inline plugins/services/src/js/containers/services/ServicesTable.js]: The code was like this already here but would be nice to refactor to ternary too.
```
const conciseOverview = tasksRunning !== instancesCount
      ? ` (${tasksRunning}/${instancesCount})`
      : ` (${tasksRunning})`;
    const verboseOverview = tasksRunning !== instancesCount
      ? ` (${tasksRunning} of ${instancesCount} Instances)`
      : ` (${tasksRunning} ${StringUtil.pluralize("Instance", tasksRunning)})`;
```

**Comment by jfurrow** [comment]
@weblancaster Thanks for the review! I implemented the ternaries you suggested, which dismissed your previous approval.

File: plugins/services/src/js/structs/ServiceTree.js [hunk 3]
```diff
@@ -388,6 +388,10 @@ module.exports = class ServiceTree extends Tree {
 
   getTaskCount() {
     return this.flattenItems().getItems().reduce(function(memo, service) {
+      if (service instanceof ServiceTree) {
```
Comment (MatApple) [inline plugins/services/src/js/structs/ServiceTree.js]: `#flattenItems` should recursively flatten the tree. Did you run in to an error or are you being extra cautious? :)

Comment (jfurrow) [inline plugins/services/src/js/structs/ServiceTree.js]: `#flattenItems` does flatten it into a single array, but the array also includes the `ServiceTree` instances, so the accumulated total was incorrect.

Comment (MatApple) [inline plugins/services/src/js/structs/ServiceTree.js]: Ah I see - you are correct. :+1:

File: plugins/services/src/js/structs/__tests__/Service-test.js [hunk 4]
```diff
@@ -32,6 +32,32 @@ describe("Service", function() {
...
+  describe("#getTaskCount", function() {
+    it("returns the number of reported tasks", function() {
+      const service = new Service({
+        tasks: [{ foo: "bar" }, { bar: "baz" }]
+      });
+
+      expect(service.getTaskCount()).toEqual(2);
+    });
+
+    it("returns the number of reported tasks", function() {
```
Comment (MatApple) [inline plugins/services/src/js/structs/__tests__/Service-test.js]: Let's rename this test to something different from the one above. It'll be easier to read the test from the test reporter to understand what it was testing and to find the exact test in code (being unique).

Perhaps `returns 0 when task array is empty`. We already have context for the method given the parent `describe` so we should name this something relevant to the test case.

Comment (jfurrow) [inline plugins/services/src/js/structs/__tests__/Service-test.js]: Good catch! I totally agree, it's fixed now!

**Pull request merged**